pcgc/basic_client.go

```diff
@@ -0,0 +1,114 @@
+package pcgc
+
+import (
+	"io"
+	"net/http"
+	"time"
+
+	"gopkg.in/errgo.v1"
+)
+
+type basicHTTPClient struct {
+	client *http.Client
+}
+
+// HTTPResponse wrapper for HTTP response objects
+type HTTPResponse struct {
+	resp *http.Response
+	err  error
+}
+
+// BasicHTTPOperation defines a contract for this client's API
+type BasicHTTPOperation interface {
+	Get(url string) HTTPResponse
```

Comment:
Why some are `*JSON` and some not?

Reply:
the JSON ones set the content-type header.